### ISSUE TITLE
Fix TypeError with `@nest` in combination with pseudo-classes with multiple arguments in selector-max-compound-selectors

### ISSUE THREAD

**Issue by woble** [stylelint/stylelint#2453]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Complex nesting via use of `@nest` (`postcss-nesting`) in combination with `:not()` and/or `:matches()` with multiple arguments produces an error.

Split from #2269

> Which rule, if any, is this issue related to?

Seems to be related to `selector-max-compound-selectors`.

> What CSS is needed to reproduce this issue?

```css
.de {
    @nest
    .cd:not(:hover, :focus) & {
        & .fg {
            color: red;
        }
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules":
  {
    "selector-max-compound-selectors": 3
  }
}
```

> Which version of stylelint are you using?

`7.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

- Use of `@nest` via `postcss-nesting`.
- Multiple arguments in `:not()`.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
Error: Expected closing parenthesis.
    at new error (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\node_modules\postcss-selector-parser\dist\processor.js:29:23)
    at Parser.error (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:234:15)
    at Parser.parentheses (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:303:22)
    at Parser.parse (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:533:22)
    at Parser.loop (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:519:18)
    at new Parser (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:104:21)
    at Processor.process (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\node_modules\postcss-selector-parser\dist\processor.js:26:21)
    at resolvedNestedSelector.forEach.resolvedSelector (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\lib\rules\selector-max-compound-selectors\index.js:76:55)
    at Array.forEach (native)
    at rule.selectors.forEach.selector (C:\Users\user\AppData\Roaming\npm\node_modules\stylelint\lib\rules\selector-max-compound-selectors\index.js:74:48)
```

**Comment by jeddy3** [comment]
@woble Thanks for creating a new issue! Much easier to digest.

It looks to be a bug. I tried `postcss-selector-parser` with `.de .cd:not(:hover, :focus) .fg` and it seemed to have no problem with it. So, it does indeed look to be a problem with the combination with `@nest`.

Labelling as `help wanted` and `bug`.

**Comment by hudochenkov** [comment]
Problem with `postcss-resolve-nested-selector`. It resolves selector as `[".de .cd:not(:hover .fg", ":focus) .de .fg"]`.

To be fair, `:not` only supports simple selectors by CSS3 specification. In working draft of CSS4 it supports complex selectors.

[Selectors Level 3](https://drafts.csswg.org/selectors-3/#negation)
[Selectors Level 4](https://drafts.csswg.org/selectors-4/#negation)

**Comment by woble** [comment]
Perhaps @davidtheclark can be of some help.

**Comment by evilebottnawi** [comment]
Seems bug in `postcss-resolve-nested-selector`

**Comment by romainmenke** [comment]
This was resolved upstream.

Verified in stylelint with:

```js
// https://github.com/stylelint/stylelint/issues/2453
testRule({
	ruleName,
	config: [3],

	accept: [
		{
			code: '.a { @nest .b:not(:hover, :focus) & { & .c {} } }',
		}
	],

	reject: [
		{
			code: '.a { @nest .b:not(:hover, :focus) & { & .c .d {} } }',
			message: messages.expected('& .c .d', 3),
			line: 1,
			column: 39,
			endLine: 1,
			endColumn: 46,
		},
	],
});
```

-----

@ybiquitous Do you prefer to close this without further action or do we update the version in `package.json` and maybe even add a test to prevent regressions?

**Comment by Mouvedia** [comment]
Do we update the CHANGELOG?

**Comment by ybiquitous** [comment]
Sorry for the late response. I just know @romainmenke got access to publish `postcss-resolve-nested-selector`! 🎉 
https://github.com/csstools/postcss-resolve-nested-selector

This is an old bug, so I think it's helpful for users to bump `postcss-resolve-nested-selector` and add a changelog entry describing this bug fix in one PR.

**Comment by ybiquitous** [comment]
I just let Dependabot to open PR #7875 to bump the package.
